### PR TITLE
Bump transformers to 4.25.1

### PR DESCRIPTION
**Pull request by katalinic-gc**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by payoto — COMMENTED** [review]
Could you bump the version to 0.6.0 at the same time?

**Comment by katalinic-gc** [comment]
> Could you bump the version to 0.6.0 at the same time?

It was bumped yesterday by James.

File: optimum/graphcore/modeling_utils.py [hunk 1]
```diff
@@ -24,12 +24,41 @@
...
 
+
+TIED_WEIGHT_MODELS = set(
+    MODEL_FOR_PRETRAINING_MAPPING.values()
+    + MODEL_FOR_MASKED_LM_MAPPING.values()
+    + MODEL_FOR_CAUSAL_LM_MAPPING.values()
+)
+
+
+def tied_weight_model(transformers_cls=None):
```
Comment (jimypbr) [inline optimum/graphcore/modeling_utils.py:47]: add a docstring for this to explain what this is for

File: tests/examples/run_clm.txt [hunk 2]
```diff
@@ -1,39 +1,37 @@
...
-51c51,52
+52c51,52
 < from transformers.utils import check_min_version, send_example_telemetry
 ---
 > from transformers.utils import check_min_version as tf_check_min_version
 > from transformers.utils import send_example_telemetry
-56c57
-< check_min_version("4.20.0")
+57c57
+< check_min_version("4.25.0")
```
[the remark this answers is not in the document]
Comment (katalinic-gc) [reply tests/examples/run_clm.txt:18]: 4.25.0; it tracks upstream transformers examples

[the remark this answers is not in the document]
Comment (katalinic-gc) [reply tests/examples/run_clm.txt:18]: https://github.com/huggingface/transformers/blob/v4.25.1/examples/pytorch/language-modeling/run_clm.py#L57

File: optimum/graphcore/modeling_utils.py (hunk 1 above)
Comment (katalinic-gc) [reply optimum/graphcore/modeling_utils.py:47]: added

**Review by jimypbr — APPROVED** [review]
great work

**Pull request merged**